### PR TITLE
[layer] Update RNN to layerv2

### PR DESCRIPTION
**Pull request by kparichay**
Update RNN to layerv2 design.
Corresponding unittests have been added and enabled.

Cleanup RNN implementation.
Reduce usage of temporary allocated memory and reuse existing memory as much as possible.
Further, request hidden state memory from context than creating by self.

Signed-off-by: Parichay Kapoor <kparichay@gmail.com>

### CODE REVIEW

File: nntrainer/layers/rnn.cpp [hunk 1]
```diff
@@ -68,32 +68,32 @@ void RNNLayer::finalize(InitLayerContext &context) {
...
+  d.width(unit);
+  wt_idx[RNNParams::hidden_state] =
+    context.requestTensor(d, "RNN:hidden_state", true, ITERATION_LIFESPAN);
 
   if (hidden_state_activation_type == ActivationType::ACT_NONE) {
     hidden_state_activation_type = ActivationType::ACT_TANH;
     acti_func.setActiFunc(hidden_state_activation_type);
   }
+
+  if (!acti_func.supportInPlace())
```
Comment (zhoonit) [inline nntrainer/layers/rnn.cpp:94]: Nice

**Review by zhoonit — APPROVED** [review]
Looks good to me :)

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nntrainer/layers/rnn.cpp [hunk 2]
```diff
@@ -149,53 +149,37 @@ void RNNLayer::exportTo(Exporter &exporter, const ExportMethods &method) const {
...
+  Tensor &hidden_ = context.getTensor(wt_idx[RNNParams::hidden_state]);
   Tensor &input_ = context.getInput(SINGLE_INOUT_IDX);
   const TensorDim &input_dim = input_.getDim();
 
-  Tensor temp;
-  Tensor hs_prev;
-  Tensor hs;
-
-  // TODO : check merge b and t index
+  // TODO: swap b and t index with transpose
```
Comment (jijoongmoon) [inline nntrainer/layers/rnn.cpp:160]: It will be great.

**Pull request merged**